File: apis/src/main/java/io/dirigible/samples/api/domain/SubExample.java
package io.dirigible.samples.api.domain;

import java.time.ZonedDateTime;

public class SubExample {

	private ZonedDateTime date;

	public ZonedDateTime getDate() {
		return date;
	}

	public void setDate(ZonedDateTime date) {
		this.date = date;
	}

	public SubExample withDate(ZonedDateTime date) {
		setDate(date);
		return this;
	}
}

File: apis/src/main/resources/META-INF/dirigible/custom-api/SubExample.ts
export class SubExample {

    // @ts-ignore
    private date: Date;

    public getDate(): Date {
        return this.date;
    }

    public setDate(date: Date): void {
        this.date = date;
    }

    public withDate(date: Date): SubExample {
        this.setDate(date);
        return this;
    }
}